### ISSUE TITLE
bug: concurrent api calls return a response that is not a quote 

### ISSUE THREAD

**Issue by CholexMax** [joswin18/MotivateMe-chrome-Extension#6]
as user if concurrently called new quote then the response api is coming no need to more request too many request need auth key for unlimited request which showing as quote to user and if some error is coming probably it gone saved in local storage as well for a response.

**Comment by mohsenkhosroanjam** [comment]
@joswin18 I would love to handle this. Would you like to assign it to me?

**Comment by joswin18** [comment]
@mohsenkhosroanjam Sure, go ahead. assigned to you!

**Comment by mohsenkhosroanjam** [comment]
@joswin18 I Just sent you a PR. 
The problem was that the rate limit error was not handled. The API responded with 200 but the response body was like a rate limit ![before](https://github.com/user-attachments/assets/905cb46c-6397-481b-950f-e04ba8c2204b)

I handled the error. So when a rate limit occurs, the user will not see a new quote and will know that a rate limit happens with a proper message. I tested it locally and also I tested the previous functionalities to ensure that my changes did not introduce new bugs. 
![after](https://github.com/user-attachments/assets/51465e2b-ff28-4f11-946f-217835229a2b)
